multi_partner_learning.py

```diff
@@ -425,22 +426,46 @@ def prepare_aggregation_weights(self):
     def aggregate_model_weights(self):
         """Aggregate model weights from the list of models, with a weighted average"""
 
-        weights_per_layer = list(zip(*self.models_weights_list))
-        new_weights = list()
+        if type(self.models_weights_list[0]) is tuple:  # Check weights type for aggregation
```

Comment:
this if/else corresponds to Logistic Regression VS neural networks ? If so, we should at least indicate it in comments

Reply:
True, I'll detail it above